spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -2344,9 +2344,6 @@ public <T> T invoke(OperationsCallback<T> action, @Nullable com.rabbitmq.client.
 				if (channel == null) {
 					throw new IllegalStateException("Connection returned a null channel");
 				}
-				if (!connectionFactory.isPublisherConfirms()) {
```

Comment:
Why remove altogether?
The discussion was about condition improvement:
```
if (!connectionFactory.isPublisherConfirms() && !connectionFactory.isSimplePublisherConfirms()) {
```

Reply:
Thanks for taking the time to review this!

I went through the discussion and thought the code and came to (maybe faulty) conclusion that closing the channel never makes sense for the `invoke` method. Especially for me this totally unexpected and not mentioned in the docs anywhere. 

Please let me know which way you would like to see this proceed!

Thanks.

Reply:
I believe the close is not required when we are in publish-confirm mode.
Therefore that condition and another PR.
Sorry, I don't see anywhere in discussion that we were going to remove this condition altogether.